pkg/crds/crds.go

```diff
@@ -0,0 +1,192 @@
+		if err != nil {
+			panic(fmt.Sprintf("Failed to read Enterprise CRD %s: %v", entry.Name(), err))
+		}
+
+		if len(yamlDelimRe.FindAllString(string(b), -1)) > 1 {
+			panic(fmt.Sprintf("Too many yaml delimiters in Enterprise CRD %s", entry.Name()))
+		}
+
+		ret[entry.Name()] = yamlDelimRe.ReplaceAll(b, []byte("\n"))
+	}
+
+	return ret
+}
+
+func getOperatorCRDSource(variant opv1.ProductVariant) map[string][]byte {
+	ret := map[string][]byte{}
+	entries, err := operatorCRDFiles.ReadDir("operator")
+	if err != nil {
+		panic(fmt.Sprintf("Failed to read Operator CRDs: %v", err))
+	}
+
+	for _, entry := range entries {
+		if variant == opv1.Calico {
+			if !calicoOprtrCRDsRe.MatchString(entry.Name()) {
+				continue
+			}
+		}
+
+		b, err := operatorCRDFiles.ReadFile(path.Join("operator", entry.Name()))
+		if err != nil {
+			panic(fmt.Sprintf("Failed to read Operator CRD %s: %v", entry.Name(), err))
+		}
+
+		if len(yamlDelimRe.FindAllString(string(b), -1)) > 1 {
+			panic(fmt.Sprintf("Too many yaml delimiters in Operator CRD %s", entry.Name()))
+		}
+
+		ret[entry.Name()] = yamlDelimRe.ReplaceAll(b, []byte("\n"))
+	}
+
+	return ret
+}
+
+func GetCRDs(variant opv1.ProductVariant) []*apiextenv1.CustomResourceDefinition {
+	lock.Lock()
+	defer lock.Unlock()
+
+	if variant == opv1.Calico {
+		if len(calicoCRDs) == 0 {
+			crdyamls := getCalicoCRDSource()
+			for _, yml := range crdyamls {
+
+				crd := &apiextenv1.CustomResourceDefinition{}
+				err := yaml.Unmarshal(yml, crd)
+				if err != nil {
+					fmt.Println(fmt.Sprintf("%s.%s", crd.Spec.Names.Plural, crd.Spec.Group))
+					fmt.Println(err)
+					return calicoCRDs
+				}
+				crd.Name = fmt.Sprintf("%s.%s", crd.Spec.Names.Plural, crd.Spec.Group)
+				calicoCRDs = append(calicoCRDs, crd)
+			}
+			crdyamls = getOperatorCRDSource(variant)
+			for _, yml := range crdyamls {
+				crd := &apiextenv1.CustomResourceDefinition{}
+				yaml.Unmarshal(yml, crd)
+				crd.Name = fmt.Sprintf("%s.%s", crd.Spec.Names.Plural, crd.Spec.Group)
+				calicoCRDs = append(calicoCRDs, crd)
+			}
+		}
+		return calicoCRDs
+	} else {
+		if len(enterpriseCRDs) == 0 {
+			crdyamls := getEnterpriseCRDSource()
+			for _, yml := range crdyamls {
+				crd := &apiextenv1.CustomResourceDefinition{}
+				yaml.Unmarshal(yml, crd)
+				crd.Name = fmt.Sprintf("%s.%s", crd.Spec.Names.Plural, crd.Spec.Group)
+				enterpriseCRDs = append(enterpriseCRDs, crd)
+			}
+			crdyamls = getOperatorCRDSource(variant)
+			for _, yml := range crdyamls {
+				crd := &apiextenv1.CustomResourceDefinition{}
+				yaml.Unmarshal(yml, crd)
+				crd.Name = fmt.Sprintf("%s.%s", crd.Spec.Names.Plural, crd.Spec.Group)
+				enterpriseCRDs = append(enterpriseCRDs, crd)
+			}
+		}
+		return enterpriseCRDs
+	}
+}
+
+// ToRuntimeObjects converts the given list of CRDs to a list of client.Objects
+func ToRuntimeObjects(crds ...*apiextenv1.CustomResourceDefinition) []client.Object {
```

Comment:
I think we already have a definition for this, but it might be in the render package (which I don't think you want to access here). You may want to consider pulling that definition out into a common package accessible by both.

Reply:
We don't have this for CustomResourceDefinitions, we have it for Secrets and/or ConfigMaps.
I don't think there is any reason that ToRuntimeObjects should go in the render package, we've just been putting it there because we've been using it for rendering. It seems good to keep it in a package that is named for what the ToRuntimeObjects operators on.

Reply:
Oh right.